### PR TITLE
Added changes for m0betool tracking all btrees

### PR DESCRIPTION
**Pull request by jamohite**

### CODE REVIEW

File: be/tool/main.c [hunk 1]
```diff
@@ -32,12 +32,29 @@
 
 #include <stdio.h>              /* printf */
 #include <stdlib.h>             /* EXIT_FAILURE */
+#include <string.h>     	/* strcmp */
```
Comment (huanghua78) [inline be/tool/main.c:35]: alignment of /* strcmp */

File: be/tool/main.c [hunk 2]
```diff
@@ -32,12 +32,29 @@
...
 
 #include "lib/string.h"         /* m0_streq */
+#include "lib/types.h"          /* PRIu64 */
+#include "lib/misc.h"           /* PRIu64 */
```
Comment (huanghua78) [inline be/tool/main.c:39]: Are these two header files all for PRIu64?

File: be/tool/main.c [hunk 3]
```diff
@@ -59,6 +78,102 @@ static const char *betool_help = ""
...
+	struct m0_buf            opt;
+	struct m0_be_seg        *seg;
+	struct m0_be_0type      *objtype;
+	struct m0_cob_domain     *cdom;
+	struct stob_ad_0type_rec *rec ;
+	struct m0_balloc         *m0balloc;
+
+	seg = m0_be_domain_seg0_get(dom);
+	if (seg == NULL)
+	{
```
Comment (huanghua78) [inline be/tool/main.c:97]: Place the opening brace at the end of previous line.

File: be/tool/main.c [hunk 4]
```diff
@@ -59,6 +78,102 @@ static const char *betool_help = ""
...
+		left > 0 ;
+		left = m0_be_segobj_opt_next(seg, objtype, &opt, &suffix)) {
+
+			M0_LOG(M0_ALWAYS, "object b0_name = '%s' suffix = '%s'"
+						"objtype = %p seg = %p",
+						objtype->b0_name, suffix,
+						objtype, seg );
+
+			if (  strcmp(objtype->b0_name, "M0_BE:COB") == 0)
+			{
```
Comment (huanghua78) [inline be/tool/main.c:112]: Please following "doc/coding-style".

File: be/tool/main.c [hunk 5]
```diff
@@ -59,6 +78,102 @@ static const char *betool_help = ""
...
+				cdom = *(struct m0_cob_domain**)opt.b_addr;
+				M0_LOG(M0_ALWAYS, "M0_BE:COB "
+						"cd_object_index btree = %p "
+						"cd_namespace btree = %p "
+						"cd_fileattr_basic btree = %p "
+						"cd_fileattr_omg btree = %p "
+						"cd_fileattr_ea btree = %p",
+						&cdom->cd_object_index,
```
Comment (huanghua78) [inline be/tool/main.c:120]: No need to print out all these addresses:
They can be calculated from "cdom", and they are consecutive.

File: be/tool/main.c [hunk 6]
```diff
@@ -59,6 +78,102 @@ static const char *betool_help = ""
...
+				if(print_btree)
+				{
+					M0_LOG(M0_ALWAYS, "em_mapping");
+					btree_dbg_print(&rec->sa0_ad_domain->
+							sad_adata.em_mapping);
+				}
+				m0balloc = container_of(
+					       rec->sa0_ad_domain->sad_ballroom,
+					       struct m0_balloc, cb_ballroom);
+				if (m0balloc != NULL)
```
Comment (huanghua78) [inline be/tool/main.c:151]: If rec->sa0_ad_domain->sad_ballroom is not NULL, the m0balloc will not be NULL.
Actually, even if rec->sa0_ad_domain->sad_ballroom is NULL, the result of container_of() will not be NULL.

File: be/tool/main.c [hunk 7]
```diff
@@ -91,6 +235,17 @@ int main(int argc, char *argv[])
...
+	if (argc > 1 && m0_streq(argv[1], "track_btree")) {
+		path = argc > 2 ? argv[2] : NULL;
+		scan(path, "track_btree");
+		return EXIT_SUCCESS;
+	}
+	if (argc > 1 && m0_streq(argv[1], "print_btree")) {
+		path = argc > 2 ? argv[2] : NULL;
+		scan(path, "print_btree");
+		return EXIT_SUCCESS;
+	}
```
Comment (huanghua78) [inline be/tool/main.c:247]: Please avoid duplicated code:
```
	if (argc > 1 && (m0_streq(argv[1], "track_btree") || m0_streq(argv[1], "print_btree"))) {
		path = argc > 2 ? argv[2] : NULL;
		scan(path, argv[1]);
		return EXIT_SUCCESS;
	}
```

**Review by huanghua78 — CHANGES_REQUESTED** [review]
some changes needed.

File: be/tool/main.c (hunk 1 above)
Comment (jamohite) [reply be/tool/main.c:35]: its correctly aligned. It is shown properly in vim, some issue with displaying here.

File: be/tool/main.c (hunk 5 above)
Comment (jamohite) [reply be/tool/main.c:120]: its required to print btree pointers in case of track_btree case

**Comment by jamohite** [comment]
created new PR by creating new branch in original repository

**Pull request closed without merging**